SuiteCRM/SuiteCRM
# WIP: Docker containers

## Description




Add support for Docker containers, this PR does/will contain a Dockerfile and matching docker-compose.yml for use in the creation and deployment of a Suite CRM container. 

It desires to be ambiguous so it can be used from the CLI to create an image or deploy a container as well as support the needs of automated deployment through a CI if needed.

At the lowest level simply running `docker build ./docker/` from the root of Suite should yield a fresh image built from the "master" branch of the official repository. 

At the highest (That I've thought of so far) using docker-compose, a user should be able to pass a target repository, branch and specify if the container should use an internal database or a separate container for the database.

No matter the level of abstraction Suite should install and set itself up, so when a container is started Suite is ready to rock.


## Motivation and Context


This breaks the dependence on third parties for the supply of containers, also will help with quick deployment of an instance.

In the running instance I am working from, gitlab i used to track any changes, but has to rsync anything done through the CI to the container that is host to suite. The container is an image from the third party bitnami (I have nothing against Bitnami, just the fact that I'm at their mercy when it comes to Suite) I've focused on the gitlab-ci to get started but this should work with any scriptable continuous integration out there. Or right from the command line.

## How To Test This


You should have both docker and docker-compose installed.

Running `docker-compose -f ./docker/docker-compose.yml up -d --force-recreate --always-recreate-deps` from the branches root will create a mariadb container and then build and deploy the docker container.

Running `docker build --force-rm -t smt/suitecrm:latest ./docker/` Will build an image that can then be used to deploy a container.

## Types of changes


### Final checklist







In summary, at this point, the concept works, if docker is available the containers get created but the silent install part fails, and the DB conficuration is incomplete.
